src/thunderbots/software/ai/hl/stp/evaluation/calc_best_shot.h

```diff
@@ -0,0 +1,97 @@
+#ifndef PROJECT_CALC_BEST_SHOT_H
+#define PROJECT_CALC_BEST_SHOT_H
+
+#include "ai/world/field.h"
+#include "ai/world/robot.h"
+#include "ai/world/world.h"
+#include "geom/point.h"
+
+namespace Evaluation
+{
+    /**
+     * Finds the length of the largest continuous interval (angle-wise) of the
+     * goal that can be seen from a point.
+     *
+     * @param f-field is needed to calculate the location of the goalposts based on field
+     * length and goal width
+     *
+     * @param obstacles-obstacles is a list of all the obstacles in the way between the
+     * point and the net
+     *
+     * @param p-the point that the shot is being calculated from
+     *
+     * @param radius-the multiplier to the radius of the robot,
+     * you can decrease the multiplier to make it easier to shoot.
+     *
+     * @return the point and the score (angle),
+     * where the score will be 0 if the point is invalid.
+     */
+    std::pair<Point, Angle> calcBestShot(const Field& f,
```

Comment:
Using the `radius` parameter as a scaling factor feels a little awkward to me. I think this would be easier to use / intuitive if this was the radius we used for the obstacles? Then the caller can decide if/how they want to scale the radius values and gives them more flexibility.

(This would apply to all the functions)

@matthewberends what do you think?

Reply:
So instead of scaling down MAX_ROBOT_RADIUS to pass to the angleCircleSweep we would directly pass in the radius received in this function?

Reply:
Changed that all functions now directly accept radius

Reply:
@matthewberends 	correct